arango/database.py

```diff
@@ -615,7 +615,10 @@ def read_log(
         search: Optional[str] = None,
         sort: Optional[str] = None,
     ) -> Result[Json]:
-        """Read the global log from server.
+        """Read the global log from server. This method is deprecated
+            in ArangoDB 3.8 and will be removed in a future version
+            of the driver. Use :func:`arango.database.Database.read_log_entries`
+            instead.
```

Comment:
Happy to adjust the way we address this for a better solution. We could for example use the `@deprecated` decorator if we add the [Deprecated](https://pypi.org/project/Deprecated/) package as a a dependency

Reply:
I think having the comment is great, but we should make it visible for users that are not going through the documentation. Or, for example, if they use another package that is built on top of `python-arango`, they should be able to figure out that there is a problem underneath.
Using the `@deprecated` decorator would do just fine. I'm not familiar with it, but a requirement that I find sensible is to allow to user to disable it the warning.
A slightly different approach that I used before is to emit a warning using the standard library, for example [here](https://github.com/ArangoDB-Community/python-arango/blob/main/arango/database.py#L2766). [FutureWarning ](https://github.com/ArangoDB-Community/python-arango/blob/main/arango/database.py#L2766) or [DeprecatedWarning](https://docs.python.org/3/library/exceptions.html#DeprecationWarning) would be fit the bill (the latter is ignored by default).

Reply:
makes sense!

I've added a `warn(..., DeprecationWarning)` call in dd7383a